File: CoreServices/src/main/java/com/example/entities/ChatMessage.java
package com.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "chat_message")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChatMessage {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    @ManyToOne
    @JoinColumn(name = "chat_room_id", nullable = false)
    private ChatRoom chatRoom;
    private UUID senderId;
    private UUID recipientId;
    private String content;
    private LocalDateTime timestamp;

    @PrePersist
    protected void onCreate() {
        timestamp = LocalDateTime.now();
    }
}

File: GUI/src/app/shared/models/chat/chat-message.ts
import { v4 as uuidv4 } from 'uuid';

export class ChatMessage {
  senderId: string;
  recipientId: string;
  content: string;
  timestamp: Date;

    constructor() {
        this.recipientId = uuidv4();
        this.senderId = uuidv4();
      }
}
